poncho/src/poncho/library_network_code.py

```diff
@@ -298,19 +354,29 @@ def main():
 
         # 5 seconds to wait for select, any value long enough would probably do
         timeout = 5
+        print_timed_message(f"set timeout for select to {timeout} seconds")
+
+        last_check_time = time.time()
 
         while True:
             # check if parent exits
             c_ppid = os.getppid()
             if c_ppid != ppid or c_ppid == 1:
+                print_timed_message("library exits with success")
                 exit(0)
 
             # wait for messages from worker or child to return
             rlist, wlist, xlist = select.select([in_pipe_fd, r], [], [], timeout)
 
+            # periodically log the number of concurrent functions
```

Comment:
Not needed. It will create many messages for information that the manager already has.